docs/tempo/website/operations/polling.md

```diff
@@ -14,17 +14,17 @@ what's called a tenant index. The tenant index is a gzip'ed json file located at
 an entry for every block and compacted block for that tenant. This is done once every `blocklist_poll` duration.
 
 All other compactors and all queriers then rely on downloading this file, unzipping it and using the contained list. 
-Again this is done once every `blocklist_poll` duration. **NOTE** It is important that the querier `blocklist_poll` duration 
-is greater than or equal to the compactor `blocklist_poll` duration. Otherwise a querier may not correctly check
-all assigned blocks and incorrectly return 404.
+Again, this is done once every `blocklist_poll` duration.
 
-Due to this behavior a given compactor or querier will often have an out of date blocklist. During normal operation
-it will stale by at most 2x the configured `blocklist_poll`. See [configuration]({{< relref "../configuration/polling" >}})
+> **NOTE** It is important that the querier `blocklist_poll` duration is greater than or equal to the compactor `blocklist_poll` duration. Otherwise a querier may not correctly check all assigned blocks and incorrectly return 404.
```

Comment:
Hmm, I was looking at the config warnings here, and I don't see one for this.  When we say querier blocklist_poll, what do we mean?

Reply:
@hedss could you answer Zach's question?

Reply:
Good question, @joe-elliott originally wrote this, and I'm not sure what it means. :)

Reply:
i'm trying to figure out who can i @ next :), but i think the buck stops here.

When we're talking about `blocklist_poll` we're talking about this value:
```
storage:
  trace:
    blocklist_poll:
```
This is basically the frequency with the component polls the backend to update its blocklist. The blocklist is currently polled by grabbing `<tenant>/index.json.gz` which is created by a tenant index builder (one of the compactors).

This means that if your `blocklist_poll` is set to `5m` on all components then the actual in memory state may be as much as `10m` old. This can happen if a querier pulls the `<tenant>/index.json.gz` the instant before the compactor recreates it.

I think the comment is wrong or possibly just outdated. If your querier/query-frontend poll frequency is much less then your compactor poll frequency then you'll repeatedly pull the same index.json.gz which is just wasted effort but won't cause query incorrectness.

There are two values I can think of that will cause query incorrectness or even query failures:
```
ingester:
  complete_block_timeout: 15m
```
This is the amount of time that an ingester will hold a complete block that it has flushed. This has to be 2x or greater than the `blocklist_poll`. This way the ingester will return traces that the queriers might not yet know about.

```
compactor:
  compaction:
    complacted_block_retention: 1h
```
This is the amount of time a compactor will let a block sit around that is ready to be deleted (b/c it's been compacted or b/c it's outside retention). Same deal as above. If this is less than 2 x block_poll then a querier might try to query a block that has been deleted.

Reply:
@joe-elliott  This is great information! Thank you. I would like to add this to the polling page in the Tempo docs.